packages/reports/src/Containers/profit-table.tsx

```diff
@@ -1,31 +1,58 @@
 import React from 'react';
 import { withRouter } from 'react-router';
 import { DesktopWrapper, MobileWrapper, DataList, DataTable } from '@deriv/components';
-import { extractInfoFromShortcode, isForwardStarting, getUnsupportedContracts, getContractPath } from '@deriv/shared';
+import {
+    extractInfoFromShortcode,
+    isForwardStarting,
+    getContractPath,
+    getSupportedContracts,
+    getUnsupportedContracts,
+} from '@deriv/shared';
 import { localize, Localize } from '@deriv/translations';
 import { ReportsTableRowLoader } from '../Components/Elements/ContentLoader';
 import CompositeCalendar from '../Components/Form/CompositeCalendar';
-import { getSupportedContracts } from '_common/contract';
+import { TInputDateRange, TColIndex, TColumnTemplateType, TContractType } from 'Types';
 
 import { connect } from 'Stores/connect';
 import EmptyTradeHistoryMessage from '../Components/empty-trade-history-message';
 import PlaceholderComponent from '../Components/placeholder-component';
 import { ReportsMeta } from '../Components/reports-meta';
 import { getProfitTableColumnsTemplate } from 'Constants/data-table-constants';
+import { TRootStore } from 'Stores/index';
+// import { ScrollParams } from 'react-virtualized';
+import moment from 'moment/moment';
 
-const getRowAction = row_obj => {
-    const contract_type = extractInfoFromShortcode(row_obj.shortcode).category.toUpperCase();
-    return getSupportedContracts()[contract_type] && !isForwardStarting(row_obj.shortcode, row_obj.purchase_time_unix)
-        ? getContractPath(row_obj.contract_id)
+type TProfitTable = {
+    component_icon: string;
+    currency: string;
+    data: Array<{ [key: string]: string }>;
+    date_from: number;
+    date_to: number;
+    error: string;
+    filtered_date_range: TInputDateRange;
+    is_empty: boolean;
+    is_loading: boolean;
+    is_switching: boolean;
+    handleDateChange: (values: { [key: string]: moment.Moment }) => void;
+    handleScroll: (ev: React.UIEvent<HTMLElement>) => void;
+    has_selected_date: boolean;
+    onMount: VoidFunction;
+    onUnmount: VoidFunction;
+    totals: React.ReactNode;
+};
+
+const getRowAction = (row_obj: { [key: string]: string }) => {
+    const contract_type = extractInfoFromShortcode(row_obj?.shortcode)?.category?.toString().toUpperCase();
+    return getSupportedContracts()[contract_type as TContractType] &&
+        !isForwardStarting(row_obj.shortcode, +row_obj.purchase_time_unix)
+        ? getContractPath(+row_obj.contract_id)
         : {
               component: (
                   <Localize
                       i18n_default_text="The {{trade_type_name}} contract details aren't currently available. We're working on making them available soon."
                       values={{
-                          trade_type_name: getUnsupportedContracts()[contract_type]?.name,
+                          trade_type_name: getUnsupportedContracts()[contract_type as TContractType].name,
```

Comment:
why did you remove this optional chaining operator?

Reply:
Not sure to be honest, added it back